svcb.go

```diff
@@ -669,6 +677,62 @@ func (s *SVCBIPv6Hint) copy() SVCBKeyValue {
 	}
 }
 
+// SVCBDoHPath pair is used to indicate the URI template that the
+// clients may use to construct a DNS over HTTPS URI.
+//
+// See RFC xxxx (https://datatracker.ietf.org/doc/html/draft-ietf-add-svcb-dns-02)
+// and RFC yyyy (https://datatracker.ietf.org/doc/html/draft-ietf-add-ddr-06).
+//
+// Basic use pattern for using the dohpath option together with an alpn
+// option:
+//
+//	s := new(dns.SVCB)
+//	s.Hdr = dns.RR_Header{Name: ".", Rrtype: dns.TypeSVCB, Class: dns.ClassINET}
+//	e := new(dns.SVCBAlpn)
+//	e.Alpn = []string{"h2", "h3"}
+//	p := new(dns.SVCBDoHPath)
+//	p.Template = "/dns-query{?dns}"
+//	s.Value = append(s.Value, e, p)
+//
+// The parsing currently only validates the length of Template.
+// It doesn't validate that Template is a valid RFC 6570 URI template.
+type SVCBDoHPath struct {
```

Comment:
“DoH” is spelled with a little “o”, because that's how [RFC 8484] spells it.  Please let me know if you find some other spelling better.

[RFC 8484]: https://datatracker.ietf.org/doc/html/rfc8484